src/net_processing.cpp

```diff
@@ -2040,8 +2040,7 @@ bool static ProcessMessage(CNode* pfrom, const std::string& strCommand, CDataStr
             // If pruning, don't inv blocks unless we have on disk and are likely to still have
             // for some reasonable time window (1 hour) that block relay might require.
             const int nPrunedBlocksLikelyToHave = MIN_BLOCKS_TO_KEEP - 3600 / chainparams.GetConsensus().nPowTargetSpacing;
-            if (fPruneMode && (!(pindex->nStatus & BLOCK_HAVE_DATA) || pindex->nHeight <= chainActive.Tip()->nHeight - nPrunedBlocksLikelyToHave))
-            {
+            if (fPruneMode && (IsBlockPruned(pindex) || pindex->nHeight <= chainActive.Tip()->nHeight - nPrunedBlocksLikelyToHave)) {
```

Comment:
nit: Explicit precedence for `chainActive.Tip()->nHeight - nPrunedBlocksLikelyToHave` would be nice

Reply:
Could drop `fPruneMode` because `fHavePruned` is only true if `fHavePruned` is also true. Same below.

Reply:
Done.